crates/red_knot_python_semantic/src/types.rs

```diff
@@ -356,6 +407,34 @@ impl<'db> Type<'db> {
             _ => Type::Unknown, // TODO type errors
         }
     }
+
+    /// Given a type that is assumed to represent an instance of a class,
+    /// return a type that represents that class itself.
+    #[must_use]
+    pub fn to_type_of_class(&self, db: &'db dyn Db) -> Type<'db> {
+        match self {
+            Type::Unbound => Type::Unbound,
+            Type::Never => Type::Never,
+            Type::Instance(class) => Type::Class(*class),
+            Type::Union(union) => union.transform(db, |ty| ty.to_type_of_class(db)),
+            Type::BooleanLiteral(_) => builtins_symbol_ty_by_name(db, "bool"),
+            Type::BytesLiteral(_) => builtins_symbol_ty_by_name(db, "bytes"),
+            Type::IntLiteral(_) => builtins_symbol_ty_by_name(db, "int"),
+            Type::Function(_) => CoreStdlibModule::Types.symbol_ty_by_name(db, "FunctionType"),
+            Type::Module(_) => CoreStdlibModule::Types.symbol_ty_by_name(db, "ModuleType"),
+            Type::None => CoreStdlibModule::Typeshed.symbol_ty_by_name(db, "NoneType"),
```

Comment:
Nit: Should we introduce helper functions like `types_symbol_ty_by_name` and `typeshed_symbol_ty_by_name` similar to `builtins_ty_by_name`?

Reply:
Could do. It depends on whether we want to keep `builtins_symbol_ty_by_name()` at all (https://github.com/astral-sh/ruff/pull/13195/files#r1740903151)